scripts/csv_bench.sh

```diff
@@ -0,0 +1,117 @@
+usage() {
+    cat <<_end_
+Usage: csv-bench.sh --path=PATH [--coreneuron]
+
+Generate CSV file summarising a set of benchmark runs.
+
+Options:
+    --path=PATH      Path containing the output
+    --coreneuron     If the path contains CoreNeuron output.
+_end_
+    exit 1
+}
+
+parse_coreneuron=false
+path=
+
+while [ "$1" != "" ]
+do
+    case $1 in
+        --coreneuron )
+            parse_coreneuron=true
+            ;;
+        --path=* )
+            path="${1#--path=}"
+            ;;
+        --path )
+            shift
+            path=$1
+            ;;
+        * )
+            echo "unknown option '$1'"
+            usage
+    esac
+    shift
+done
+
+if [ "$path" == "" ]
+then
+    usage
+    exit 1
+fi
+
+if [ ! -d "$path" ]
+then
+    echo "error: path \"$path\" does not exist"
+    exit 1
+fi
+
+table_line() {
+    fid="$1"
+    line=
+
+    tts=`awk '/^model-run/ {print $2}' "$fid"`
+    ncell=`awk '/^cell stats/ {print $3}' "$fid"`
+
+    line="$(printf "%9d,%12.3f," $ncell $tts)"
+    nranks=`awk '/^ranks:/ {print $2}' "$fid"`
+
+    mempos=`awk '/^meter / {j=-1; for(i=1; i<=NF; ++i) if($i =="memory(MB)") j=i; print j}' "$fid"`
+    if [ "$mempos" != "-1" ]
+    then
+        rankmem=$(awk "/^meter-total/ {print \$$mempos}" "$fid")
+        totalmem=`echo $rankmem*$nranks | bc -l`
+        line="$line$(printf "%12.3f," $totalmem)"
+    else
+        line="$line$(printf "%12s," "")"
+    fi
+
+    nthreads=`awk '/^threads:/ {print $2}' "$fid"`
+    hasgpu=`awk '/^gpu:/ {print $2}' "$fid"`
+    line="$line$(printf "%7d,%7d,%7s" $nranks $nthreads $hasgpu)"
+}
+
+# NOTE: this is very fragile and will almost certainly break from version to
+# version of CoreNeuron. There is not much we can do about that, because the
+# only information we have available is whatever CoreNeuron outputs to stdout.
+table_line_cnr() {
+    fid="$1"
+    line=
+
+    tts=`grep "Solver Time" "$fid" | awk '{print $4}'`
+    ncell=`grep "Number of cells" "$fid" | awk '{print $4}'`
+
+    rankmem_start=`grep "After MPI_Init" "$fid" | awk '{print $12}'`
+    rankmem_end=`grep "After nrn_finitialize" "$fid" | awk '{print $12}'`
+    rankmem=`echo $rankmem_end-$rankmem_start | bc -l`
+    # num_mpi and num_omp_thread are printed more than once (for redundancy?)
+    # So use awk to discard all but the last occurence of each variable
+    nranks=`awk '/num_mpi/ {split($1,line,"="); x=line[2]} END{print x}' "$fid"`
+    nthreads=`awk '/num_omp_thread/ {split($1,line,"="); x=line[2]} END{print x}' "$fid"`
```

Comment:
Nothing wrong with this, but could consider using the field separator to simplify this, e.g. `awk -F= '/num_mpi/ {x=$2} END {print x}'`

Reply:
I didn't know about this one. I have changed to use `-F=`, but I note that I could have done things more cleanly with the original approach `/num_mpi/ {split($1,x,"=")} END{print x}`.